codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -1060,8 +1038,8 @@ void removeFromActiveStreams(DefaultStream stream) {
                         listeners.get(i).onStreamClosed(stream);
                     }
                 } finally {
-                    // Mark this stream for removal.
-                    removalPolicy.markForRemoval(stream);
+                    // Remove the stream entirely if it has no children in the priority tree.
```

Comment:
@nmittler I think this comment is not right. Streams should be removed even if they have children, but the children should be reprioritised and get a new parent?